packages/test/src/test-worker-deployment-versioning.ts

```diff
@@ -0,0 +1,409 @@
+});
+
+test('Workflows can use default versioning behavior', async (t) => {
+  const taskQueue = 'task-queue-default-versioning-' + randomUUID();
+  const deploymentName = 'deployment-default-versioning-' + randomUUID();
+  const { client, nativeConnection } = t.context.env;
+
+  const workerV1 = {
+    buildId: '1.0',
+    deploymentName,
+  };
+
+  const worker = await Worker.create({
+    workflowsPath: require.resolve('./deployment-versioning-no-annotations'),
+    taskQueue,
+    workerDeploymentOptions: {
+      useWorkerVersioning: true,
+      version: workerV1,
+      defaultVersioningBehavior: 'PINNED',
+    },
+    connection: nativeConnection,
+  });
+
+  const workerPromise = worker.run();
+  workerPromise.catch((err) => {
+    t.fail('Worker run error: ' + JSON.stringify(err));
+  });
+
+  const describeResp = await waitUntilWorkerDeploymentVisible(client, workerV1);
+  await setCurrentDeploymentVersion(client, describeResp.conflictToken, workerV1);
+
+  const wf = await client.workflow.start('noVersioningAnnotationWorkflow', {
+    taskQueue,
+    workflowId: 'default-versioning-behavior-' + randomUUID(),
+  });
+
+  await wf.result();
+
+  const history = await wf.fetchHistory();
+  const hasPinnedVersioningBehavior = history.events!.some(
+    (event) =>
+      event.workflowTaskCompletedEventAttributes &&
+      event.workflowTaskCompletedEventAttributes.versioningBehavior ===
+        temporal.api.enums.v1.VersioningBehavior.VERSIONING_BEHAVIOR_PINNED
+  );
+  assert.ok(hasPinnedVersioningBehavior, 'Expected workflow to use pinned versioning behavior');
+
+  worker.shutdown();
+  await workerPromise;
+  t.pass();
+});
+
+async function setRampingVersion(
+  client: Client,
+  conflictToken: Uint8Array,
+  version: WorkerDeploymentVersion,
+  percentage: number
+) {
+  return await client.workflowService.setWorkerDeploymentRampingVersion({
+    namespace: client.options.namespace,
+    deploymentName: version.deploymentName,
+    version: toCanonicalString(version),
+    conflictToken,
+    percentage,
+  });
+}
+
+async function waitUntilWorkerDeploymentVisible(client: Client, version: WorkerDeploymentVersion) {
+  return await asyncRetry(
+    async () => {
+      const resp = await client.workflowService.describeWorkerDeployment({
+        namespace: client.options.namespace,
+        deploymentName: version.deploymentName,
+      });
+      const isVersionVisible = resp.workerDeploymentInfo!.versionSummaries!.some(
+        (vs) => vs.version === toCanonicalString(version)
+      );
+      if (!isVersionVisible) {
+        throw new Error('Version not visible yet');
+      }
+      return resp;
+    },
+    { maxTimeout: 1000, retries: 10 }
+  );
+}
+
+async function setCurrentDeploymentVersion(
+  client: Client,
+  conflictToken: Uint8Array,
+  version: WorkerDeploymentVersion
+) {
+  return await client.workflowService.setWorkerDeploymentCurrentVersion({
```

Comment:
Note that we'll need high level Client API support for these before the feature moves to GA. It's totally fine waiting for later PR.

I think the high level API should hide the conflict token, e.g. they would pass in a lambda function; the client would do a loop of (describe, call user function, setWorkerDeplCurVersion) until the update passes.

Reply:
Yeah, that's how it will work, but yes saved for later